crates/red_knot_python_semantic/src/semantic_index/use_def.rs

```diff
@@ -704,30 +723,30 @@ impl<'db> UseDefMapBuilder<'db> {
     }
 
     /// Restore the current builder symbols state to the given snapshot.
-    pub(super) fn restore(&mut self, snapshot: FlowSnapshot) {
+    pub(super) fn restore(&mut self, snapshot: &FlowSnapshot) {
```

Comment:
I prefer accepting own instances for functions that unconditionally clone the parameter to make the cost more visible to callers. It can also help avoiding unnecessary calls in the cases where doesn't use the owned instance besides calling the function